mayastor/src/subsys/nvmf/subsystem.rs

```diff
@@ -471,6 +472,35 @@ impl NvmfSubsystem {
         }
     }
 
+    /// set ANA state: optimized, non_optimized, inaccessible
+    /// subsystem must be in paused or inactive state
+    pub async fn set_ana_state(&self, ana_state: u32) -> Result<(), Error> {
+        extern "C" fn set_ana_state_cb(arg: *mut c_void, status: i32) {
+            let s = unsafe { Box::from_raw(arg as *mut oneshot::Sender<i32>) };
+            s.send(status).unwrap();
+        }
+        let cfg = Config::get();
+        let trid_replica = TransportId::new(cfg.nexus_opts.nvmf_replica_port);
+
+        let (s, r) = oneshot::channel::<i32>();
+
+        unsafe {
+            nvmf_subsystem_set_ana_state(
+                self.0.as_ptr(),
+                trid_replica.as_ptr(),
+                ana_state,
+                Some(set_ana_state_cb),
+                cb_arg(s),
+            );
+        }
+
+        r.await.unwrap().to_result(|e| Error::Subsystem {
+            source: Errno::from_i32(e),
```

Comment:
It looks like we should use -e here, as Errno::from_i32() expects a positive errno value when translating it into Errno, whereas nvmf_subsystem_set_ana_state() passes negative integers to the callback.
```
ctx = calloc(1, sizeof(*ctx));                                                                                |{
        if (!ctx) {                                                                                                   |        struct spdk_bdev_channel *channel = spdk_io_channel_get_ctx(ch);
                SPDK_ERRLOG("Unable to allocate context\n");                                                          |
                cb_fn(cb_arg, -ENOMEM);                                                                               |        *stat = channel->stat;
                return;                                                                                               |}
        }
```

Reply:
Good point, I've even fixed a couple of instances of that in this file. I'll fix this instance but I think this affects other users in the codebase as well so I'll raise a ticket for that.